src/stores/notificationStore.ts

```diff
@@ -0,0 +1,49 @@
+import { action, computed, observable } from 'mobx';
+
+interface INotification {
+    message: string;
+    type: string;
+}
+
+export class NotificationStore {
+    @observable private _notifications: INotification[];
+
+    constructor() {
+        this._notifications = [];
+    }
+
+    @computed
+    get notifications(): INotification[] {
+        return this._notifications;
+    }
+
+    @action
+    public addNotifications(notifications: INotification[]) {
+        const currentMessages = this._notifications.map((notification: INotification) => {
+            return notification.message;
+        });
+
+        const newNotifications = notifications.filter((notification: INotification) => {
+            return (!currentMessages.includes(notification.message));
+        });
+
+        this._notifications = this._notifications.concat(newNotifications);
```

Comment:
Can this be said just with:
```
public addNotification(notification: INotification) {
    if (this._notifications.some(n => n.message === notification.message)) {
            this._notifications.push(notification);
    }
}
```

Reply:
https://stackoverflow.com/questions/8217419/how-to-determine-if-javascript-array-contains-an-object-with-an-attribute-that-e